app/assets/javascripts/test/tracing-view-screenshot-tests/dataset_rendering.screenshot.js

```diff
@@ -0,0 +1,124 @@
+/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
+const test = require("ava");
+const puppeteer = require("puppeteer");
+const pixelmatch = require("pixelmatch");
+const { PNG } = require("pngjs");
+const fs = require("fs");
+const path = require("path");
+const { screenshotDataset, DEV_AUTH_TOKEN } = require("./dataset_rendering_helpers");
+
+process.on("unhandledRejection", (err, promise) => {
+  console.error("Unhandled rejection (promise: ", promise, ", reason: ", err, ").");
+});
+
+const BASE_PATH = path.join(__dirname, "../screenshots");
+
+let BRANCH = "master";
+if (!process.env.BRANCH) {
+  console.warn(
+    "[Warning] No branch specified, assuming branch master. If you want to specify a branch, prepend BRANCH=<branchname> to the command.",
+  );
+} else {
+  BRANCH = process.env.BRANCH;
+  console.log(`[Info] Executing tests on branch ${BRANCH}.`);
+}
+
+function compareScreenshot(screenshotBuffer, name, width, height) {
+  return new Promise(resolve => {
+    let readFiles = 0;
+    const existingScreenshot = fs
+      .createReadStream(`${BASE_PATH}/${name}.png`)
+      // If there is no existing screenshot, save the current one
+      .on("error", saveScreenshot)
```

Comment:
Maybe unrealistic, but this would also assume that there is no screenshot, if there is a problem while parsing the file.

Reply:
I've added a better error check here :)